ui/components/src/Table/Table.tsx

```diff
@@ -47,6 +56,40 @@ export function Table<TableData>({
     onRowSelectionChange?.(newRowSelection);
   };
 
+  const handleRowSelectionEvent = useCallback(
+    (table: TanstackTable<TableData>, row: Row<TableData>, isModified: boolean) => {
+      if (rowSelectionVariant === 'standard' || isModified) {
+        row.toggleSelected();
+      } else {
+        // Legend variant (when action not modified with shift/meta key).
+        // Note that this behavior needs to be kept in sync with behavior in
+        // the Legend component for list-based legends.
```

Comment:
There really isn't a great way to use a single util for this because the table and legend list have some different interaction patterns.